src/Library/demos/Launcher/main.vala

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env -S vala workbench.vala --pkg libadwaita-1
+
+private Gtk.FileLauncher file_launcher;
+private Gtk.Entry uri_details;
+
+public void main () {
+  var launch_file = (Gtk.Button) workbench.builder.get_object ("launch_file");
+  var change_file = (Gtk.Button) workbench.builder.get_object ("change_file");
+  var file_location = (Gtk.Button) workbench.builder.get_object ("file_location");
+  var file_name = (Gtk.Label) workbench.builder.get_object ("file_name");
+  uri_details = (Gtk.Entry) workbench.builder.get_object ("uri_details");
+  var uri_launch = (Gtk.Button) workbench.builder.get_object ("uri_launch");
+
+  // File Launcher
+  var file = File.new_for_uri (workbench.resolve ("workbench.txt"));
+  file_launcher = new Gtk.FileLauncher (file) {
+    always_ask = true
+  };
+
+  launch_file.clicked.connect (on_launch_file);
+  file_launcher.notify["file"].connect (() => {
+    try {
+      FileInfo info = file_launcher.file.query_info ("standard::display-name", NONE, null);
+      file_name.label = info.get_display_name ();
+    } catch (Error e) {
+      critical (e.message);
+    }
+  });
+
+  file_location.clicked.connect (open_file_location);
```

Comment:
`open_file_location.begin`

And all the other async functions connect's also with `.begin`